backend/de.metas.business/src/main/java/de/metas/costrevaluation/callout/M_CostRevaluation.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * #%L
+ * de.metas.business
+ * %%
+ * Copyright (C) 2022 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package de.metas.costrevaluation.callout;
+
+import de.metas.costrevaluation.ICostRevaluationDAO;
+import de.metas.document.DocTypeId;
+import de.metas.document.IDocTypeDAO;
+import de.metas.document.sequence.IDocumentNoBuilderFactory;
+import de.metas.document.sequence.impl.IDocumentNoInfo;
+import de.metas.util.Services;
+import lombok.NonNull;
+import org.adempiere.ad.callout.annotations.Callout;
+import org.adempiere.ad.callout.annotations.CalloutMethod;
+import org.compiere.model.I_C_DocType;
+import org.compiere.model.I_M_CostRevaluation;
+
+@Callout(I_M_CostRevaluation.class)
+public class M_CostRevaluation
+{
+	@CalloutMethod(columnNames = I_M_CostRevaluation.COLUMNNAME_C_DocType_ID)
+	public void onDocTypeChanged(@NonNull final I_M_CostRevaluation costRevaluation)
```

Comment:
the logic from this method kind-of does not make sense
i.e. this method is called when user is changing M_CostRevaluation.C_DocType_ID. 
And in that case, this method *ignores* what user selected and searches for a default doctype and use it to generate the documentno.

Instead, i suggest to:
* let this class implement ITabCallout (don't forget to register it) and implement `onNew` method. There you can call the BL which gives you the default doc type and set it
* here, in `onDocTypeChanged` method, just use the M_CostRevalulation.C_DocType_ID user selected it. In case it's not set, simply `return;`